### PR TITLE
Initial start for prototype pollution prevention cheatsheet

### PR DESCRIPTION
**Pull request by tghosth**

### CODE REVIEW

**Review by jmanico — DISMISSED** [review]
Looking good so far, you can also start in the staging directory if you want more rapid commits https://github.com/OWASP/CheatSheetSeries/tree/master/cheatsheets_draft

**Comment by tghosth** [comment]
Thanks @jmanico, I think this will do for now and then people add to it as they see fit. The important thing is that the information was captured.

**Comment by tghosth** [comment]
@jmanico can you now merge it?

**Comment by tghosth** [comment]
Any chance this can now be merged @szh @jmanico ?

File: cheatsheets/Prototype_Pollution_Prevention_Cheat_Sheet.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+let obj = {__proto__:null};
+```
+
+### Use object "freeze" and "seal" mechanisms
+
+You can also use the `Object.freeze()` and `Object.seal()` APIs to prevent built-in prototypes from being modified however this can break the application if the libraries they use modify the built-in prototypes.
+
+### Node.js configuration flag
+
+Node.js also offers the ability to remove the `__proto__` property completely using the `--disable-proto=delete` flag. Note this is a defence in depth measure.
```
Comment (szh) [inline cheatsheets/Prototype_Pollution_Prevention_Cheat_Sheet.md:45]: Let's keep spellings consistent with US English
```suggestion
Node.js also offers the ability to remove the `__proto__` property completely using the `--disable-proto=delete` flag. Note this is a defense in depth measure.
```

File: cheatsheets/Prototype_Pollution_Prevention_Cheat_Sheet.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Prototype pollution is still possible using constructor.prototype properties but removing `__proto__` helps reduce attack surface and prevent certain attacks.
```
Comment (szh) [inline cheatsheets/Prototype_Pollution_Prevention_Cheat_Sheet.md:47]:
```suggestion
Prototype pollution is still possible using `constructor.prototype` properties but removing `__proto__` helps reduce attack surface and prevent certain attacks.
```

**Review by szh — CHANGES_REQUESTED** [review]
Two nits. Also should we include examples of what not to do, and show example exploits? Even just one simple example would be helpful for understanding the danger.

File: cheatsheets/Prototype_Pollution_Prevention_Cheat_Sheet.md (hunk 2 above)
Comment (tghosth) [reply cheatsheets/Prototype_Pollution_Prevention_Cheat_Sheet.md:47]: Fixed in #1161

**Pull request merged**